src/widetable/executor.py

```diff
@@ -173,7 +177,17 @@ def spja_query(self,
         if len(group_by) > 0:
             sql += "GROUP BY " + ",".join(group_by) + '\n'
         if len(order_by) > 0:
-            sql += 'ORDER BY ' + ",".join(order_by) + '\n'
+            # generate order by query with preference given to custom_order_pref values using CASE WHEN
+            order_by_query = 'ORDER BY '
+            if len(custom_order_pref) > 0:
+                for i, col in enumerate(order_by):
+                    order_by_query += f"CASE WHEN {col} = '{custom_order_pref[i]}' THEN 0 ELSE 1 END, {col}"
+                    if i <= len(order_by) - 2:
+                        order_by_query += ', '
+                sql += order_by_query + '\n'
+            else:
+                sql += "ORDER BY " + ",".join(order_by) + '\n'
```

Comment:
This CASE WHEN doesn't seem to be necessary. Can you have use https://duckdb.org/docs/sql/functions/char.html?

I know the goal is to first choose a specified join key on the top, then sort the rest.
Does a simple [ORDER BY {col} = "XX", col] work?

Reply:
I did not see any syntax like that on duckdb documentation (https://duckdb.org/docs/sql/query_syntax/orderby.html). I'll try it out tonight. Will also go through the sql functions it provides and check.